src/engine.cc

```diff
@@ -0,0 +1,83 @@
+#include "engine.h"
+#include "gobject.h"
+
+#include <eos-knowledge-content.h>
+
+using namespace v8;
+
+struct CallbackData {
+  Persistent<Promise::Resolver> resolver;
+};
+
+static void EngineGetObjectCallback(GObject *source,
+                                    GAsyncResult *result,
+                                    gpointer user_data)
+{
+  Isolate *isolate = Isolate::GetCurrent ();
+  HandleScope scope(isolate);
+  CallbackData *data = (CallbackData *)user_data;
+
+  GError *error = NULL;
+  EkncContentObjectModel *model;
+
+  Local<Promise::Resolver> resolver = Local<Promise::Resolver>::New(isolate, data->resolver);
+  if ((model = eknc_engine_get_object_finish (EKNC_ENGINE (source), result, &error)))
+    resolver->Resolve(WrapperFromGObject (isolate, G_OBJECT (model)));
+  else
+    resolver->Reject(Exception::TypeError(String::NewFromUtf8(isolate, error->message)));
+
+  /* XXX: For some reason we have to resolve microtasks to the promise wrappers to resolve. */
```

Comment:
Probably can remove this comment, here and below. I've seen this [here](https://github.com/nodejs/node/issues/5691#issuecomment-196277918):
> It also seems that it is implied by v8 authors that you should run microtasks manually when you work with promises from C++.

Although I've been googling for a "citation needed" for that and can't find anything.